lib/celluloid/io/stream.rb

```diff
@@ -80,6 +71,20 @@ def syswrite(string)
         total_written
       end
 
+      # TODO: remove after ending ruby 2.0.0 support
+      if RUBY_VERSION >= "2.1" 
+        def read_nonblock(*args, **options)
```

Comment:
You *are* redefining it under Ruby 2.1+ such that:

* The `exception: ...` parameter is completely ignored
* The default is switched from exceptional to exceptionless